packages/scheduler/lib/workers/scheduling/scheduling.worker.ts

```diff
@@ -116,11 +116,16 @@ export class SchedulingChild {
                             createdToStartedTimeoutSecs: schedule.createdToStartedTimeoutSecs,
                             startedToCompletedTimeoutSecs: schedule.startedToCompletedTimeoutSecs,
                             heartbeatTimeoutSecs: schedule.heartbeatTimeoutSecs
-                        });
-                        if (task.isErr()) {
-                            logger.error(`Failed to create task for schedule: ${schedule.id}`);
+                        })
+                    );
+                    const res = await Promise.allSettled(createTasks);
+                    for (const taskRes of res) {
+                        if (taskRes.status === 'rejected') {
+                            logger.error(`Failed to schedule task: ${taskRes.reason}`);
+                        } else if (taskRes.value.isErr()) {
+                            logger.error(`Failed to schedule task: ${stringifyError(taskRes.value.error)}`);
                         } else {
-                            taskIds.push(task.value.id);
+                            taskIds.push(taskRes.value.value.id);
```

Comment:
a bit weird but the PromiseAllSettled result type is `PromiseSettleResult<Result<Task>>[]`